src/lib.rs

```diff
@@ -184,52 +315,567 @@ decl_module! {
+            }
+        }
+
+        updated_values.append(&mut new_values);
+
+        let mut updated_values_with_nones = updated_values.clone();
+        for &id in schema_prop_ids.iter() {
+    
+            // If value was not povided for schema prop:
+            if updated_values.iter().find(|prop| prop.in_class_index == id).is_none() {
+                
+                let class_prop = class.properties.get(id as usize).unwrap();
+
+                // Check that all required prop values are provided
+                if class_prop.required {
+                    return Err(ERROR_MISSING_REQUIRED_PROP)
+                } else {
+                    // Add all missing non required schema prop values as PropertyValue::None
+                    let none_prop_value = ClassPropertyValue {
+                        in_class_index: id,
+                        value: PropertyValue::None
+                    };
+                    updated_values_with_nones.push(none_prop_value);
+                }
+            }
+        }
+
+        <EntityById<T>>::mutate(entity_id, |entity| {
+            entity.in_class_schema_indexes.push(schema_id);
+            entity.values = updated_values_with_nones;
+        });
+
+        Self::deposit_event(RawEvent::EntitySchemaAdded(entity_id));
+        Ok(())
+    }
+
+    pub fn update_entity_properties(
+        entity_id: EntityId,
+        new_property_values: Vec<ClassPropertyValue>
+    ) -> dispatch::Result {
+
+        Self::ensure_known_entity_id(entity_id)?;
+
+        let (entity, class) = Self::get_entity_and_class(entity_id);
+
+        ensure!(!entity.in_class_schema_indexes.is_empty(), ERROR_ENTITY_DOES_NOT_SUPPORT_SCHEMAS_YET);
+
+        let mut updated_values = entity.values;
+
+        for new_prop_value in new_property_values.iter() {
+            let ClassPropertyValue {
+                in_class_index: id,
+                value: new_value
+            } = new_prop_value;
+            if let Some(prop) = updated_values.iter_mut().find(|prop| *id == prop.in_class_index) {
+                let ClassPropertyValue {
+                    in_class_index: valid_id,
+                    value: current_value
+                } = prop;
+                let class_prop = class.properties.get(*valid_id as usize).unwrap();
+
+                Self::ensure_property_value_is_valid(new_value.clone(), class_prop.clone())?;
+
+                *current_value = new_value.clone();
+            } else {
+                return Err(ERROR_UNKNOWN_ENTITY_PROP_ID)
+            }
+        }
+
+        <EntityById<T>>::mutate(entity_id, |entity| {
+            entity.values = updated_values;
+        });
+
+        Self::deposit_event(RawEvent::EntityPropertiesUpdated(entity_id));
+        Ok(())
+    }
+    
+    /// Only non required property values can be removed.
+    /// In fact when removing a property value, it is replaced with PropertyValue::None.
+    pub fn remove_entity_properties(
```

Comment:
First, given that this does in fact not remove properties from entities, but instead sets property values to be none/not-set, calling it `remove_entity_properties` is not a great name.

Second, why is this needed, can't one just use `update_entity_properties` for this?

Reply:
Function `remove_entity_properties` deleted here https://github.com/Joystream/substrate-versioned-store-module/pull/16/commits/e6e62165e1ae5e64123f485f5e2d4141d020a5c2